providers/reddit/reddit.go

```diff
@@ -16,5 +17,13 @@ type Provider struct{}
 
 // BuildURI generates a search URL for Reddit.
 func (p *Provider) BuildURI(q string) string {
+	// If q starts with "r/subreddit query", treat as subreddit and do restrict
+	// search on the subreddit.
+	if len(q) > 3 && q[:3] == "/r/" {
+		subreddit := strings.Split(q, " ")[0]
+		return fmt.Sprintf("https://www.reddit.com%s/search?q=%s"+
+			"&restrict_sr=on&sort=relevance&t=all",
+			subreddit, url.QueryEscape(q[len(subreddit)+1:]))
```

Comment:
Since you split before, if you store the full split version, then you can have the subreddit and the remaining string. Then in url.QueryEscape you can use strings.Join(val[1:]...).